src/python.rs

```diff
@@ -481,7 +481,26 @@ impl Taxonomy {
                     "Node can not be moved to its child",
                 ));
             }
+
+            let old_parent_idx = self.tax.parent_ids[idx];
+            let new_parent_idx = py_try!(self.tax.to_internal_index(p));
+
+            // update id to parent_id
             self.tax.parent_ids[idx] = py_try!(self.tax.to_internal_index(p));
+
+            // remove current node from children lookup for old parent
+            // TODO: assert that exactly one item was removed?
+            // assumes values are sorted!
+            match self.tax.children_lookup[old_parent_idx].binary_search(&idx) {
```

Comment:
I think we can unwrap/expect the binary search here, we should never be in case where there is a `self.tax.parent_ids[idx]` but `idx` is not found in `self.tax.children_lookup[old_parent_idx]`. That would mean we failed to index properly and it would better to blow up so it is visible

Reply:
Updated.